docs/reference/data-streams/change-mappings-and-settings.asciidoc

```diff
@@ -0,0 +1,520 @@
+
+[source,console]
+----
+PUT /_index_template/new_logs_data_stream
+{
+  "index_patterns": [ "new_logs*" ],
+  "data_stream": {
+    "timestamp_field": "@timestamp"
+  },
+  "template": {
+    "mappings": {
+      "properties": {
+        "@timestamp": {
+          "type": "date_nanos"                 <1>
+        }
+      }
+    },
+    "settings": {
+      "index.lifecycle.name": "logs_policy",
+      "sort.field" : [ "@timestamp"],          <2>
+      "sort.order" : [ "desc"]                 <3>
+    }
+  }
+}
+----
+// TEST[continued]
+
+<1>  Changes the `@timestamp` field mapping to the `date_nanos` field datatype.
+<2>  Adds the `sort.field` index setting.
+<3>  Adds the `sort.order` index setting.
+====
+
+. Use the <<indices-create-data-stream,create data stream API>> to manually
+create the new data stream. The name of the data stream must match the name or
+wildcard pattern defined in the new template's `index_patterns` property.
++
+IMPORTANT: If you do not want to mix new and old data, do not use an indexing
+request to create the data stream. Mixing old and new data is safe, but the
+backing index containing this data needs to be retained until you are ready to
+delete the new data.
++
+.*Example*
+[%collapsible]
+====
+The following create data stream API request targets `new_logs`, which matches
+the wildcard pattern for the `new_logs_data_stream` template. Because no
+existing index or data stream uses this name, this request creates the
+`new_logs` data stream.
+
+[source,console]
+----
+PUT /_data_stream/new_logs
+----
+// TEST[continued]
+====
+
+. If you do not want to mix new and old data in your new data stream, pause the
+indexing of new documents.
+
+. If you use {ilm-init} to <<getting-started-index-lifecycle-management,automate
+rollover>>, reduce the {ilm-init} poll interval to ensure that the current write
+index doesn’t grow too large while waiting for the rollover check. By default,
+{ilm-init} checks rollover conditions every 10 minutes.
++
+.*Example*
+[%collapsible]
+====
+The following <<cluster-update-settings,update cluster settings API>> request
+lowers the `indices.lifecycle.poll_interval` setting to `1m` (one minute).
+
+[source,console]
+----
+PUT /_cluster/settings
+{
+  "transient": {
+    "indices.lifecycle.poll_interval": "1m"
+  }
+}
+----
+// TEST[continued]
+
+////
+[source,console]
+----
+DELETE /_data_stream/logs
+
+DELETE /_data_stream/new_logs
+
+DELETE /_index_template/logs_data_stream
+
+DELETE /_index_template/new_logs_data_stream
+
+DELETE /_ilm/policy/logs_policy
+----
+// TEST[continued]
+////
+====
+
+. Reindex your data to the new data stream using an `op_type` of `create`.
++
+If you want to partition the data in the order in which it was originally
+indexed, you can run separate reindex requests. These reindex requests can use
+individual backing indices as the source. You can use the
+<<indices-get-data-stream,get data stream API>> to retrieve a list of backing
+indices.
```

Comment:
Do we want to promote this concept? If we talked about ILM managing the data stream (which presumably has a rollover policy) do we want to talk about backing indices? (which are an implementation details for data streams)

Reply:
I think we have to strike a careful balance between completely hiding the fact that data streams include multiple backing indices (which I think would be unhelpful) and always talking about operating on individual backing indices (which would eliminate much of the value of data streams).

In this particular case, I can see scenarios where using individual backing indices as reindex sources would be useful. E.g., when you have a data stream with a large number of backing indices and you want to be able to be able to easily resume the reindex process if it fails at any point during the operation. Perhaps in the future we could make the reindex operation handle resuming operations on data streams without needing to reference individual backing indices, but that would certainly be an enhancement for a future phase.